src/Microsoft.TestPlatform.Common/DataCollection/DataCollectorInformation.cs

```diff
@@ -176,7 +175,7 @@ internal void DisposeDataCollector()
         /// <summary>
         /// The get test execution environment variables sync.
         /// </summary>
-        private void SetTestExecutionEnvironmentVariables()
+        public void SetTestExecutionEnvironmentVariables()
```

Comment:
We are delaying getting Environment variables for datacollectors. If we do it early(that is before initializing all DataCollectors) then it creates a partial collection plan, which contains info about only one(current) datacollector.